bootstrap/helpers/parsers.php

```
@@ -16,6 +16,132 @@
+/**
+ * Validates a Docker Compose YAML string for command injection vulnerabilities.
+ * This should be called BEFORE saving to database to prevent malicious data from being stored.
+ *
+ * @param  string  $composeYaml  The raw Docker Compose YAML content
+ *
+ * @throws \Exception If the compose file contains command injection attempts
+ */
+function validateDockerComposeForInjection(string $composeYaml): void
+{
+    try {
+        $parsed = Yaml::parse($composeYaml);
```

PHPMD (2.15.0)

Avoid using static access to class '\Symfony\Component\Yaml\Yaml' in method 'validateDockerComposeForInjection'. (undefined)

(StaticAccess)